### ISSUE TITLE
Simple browser/JavaScript demo

### ISSUE THREAD

**Issue by josephrocca** [chengtan9907/SimVPv2#2]
Just thought I'd let people know that I created a simple web/browser demo using ONNX Runtime Web.

Inference takes about 2 seconds for the 10-frame moving mnist model using the Wasm backend. It would probably be significantly faster with the WebGL backend, but [it lacks quantization op support](https://github.com/microsoft/onnxruntime/issues/13800).

The quantized model comes in at less than 50mb, and seems to match the accuracy of the non-quantized (>150mb) model.

Here's the code and demo:
* https://github.com/josephrocca/SimVP-web
* https://josephrocca.github.io/SimVP-web/demo/simple.html

My next step is to create a version of the [training and conversion notebook](https://colab.research.google.com/gist/josephrocca/80c4a9814621d228dfb3d262a26a0806) that can train on arbitrary videos that the user uploads.

(I'm also wondering whether some sort of diffusion-like process could be used to prevent the increasing blurriness as we predict further into the future. Maybe an "un-diffusion" process could actually just be "embedded" as extra frames between the "actual" frames during training? Or maybe it'd need to be a separate model. If anyone has any thoughts on this I'd love to hear them)

Thanks to the paper authors for publishing and open sourcing this!

**Comment by josephrocca** [comment]
Hmm - wondering if it's possible for you to share your video data loading code e.g. for `kitticaltech`?

**Comment by josephrocca** [comment]
I'm probably mostly talking to myself here :grin: but I managed to get video loading working and trained a proof-of-concept model on a minions movie for just a few epochs:

* https://colab.research.google.com/gist/josephrocca/7be01186b14e07b5d54e80c245a35570
* https://josephrocca.github.io/SimVP-web/demo/rise-of-gru.html

My video dataloader code is really slow (abotu a tenth of the speed of MMNIST data loader), and `num_workers>0` gives a segmentation fault. Going to keep trying to work out what's going on, but it would be super handy if there is already some code laying around somewhere for efficient video loading.

**Comment by chengtan9907** [comment]
Sorry for the late reply. I'm so glad to see our code can somewhat help you. I believe some diffusion-like things can help to improve the performance, it will be a meaningful direction for future research. It would be great to cooperate with you on such things.

**Comment by Lupin1998** [comment]
Hi, @josephrocca, thanks for your wonderful website demo. Currently, we are working on this project to add more benchmark results (`e.g.,` benchmarking `MetaFormer` with the SimVP baseline) and video prediction datasets (e.g., adding `kitticaltech`). We also plan to release trained models and logs in the future in regard to the convenience of the community. If you are still interested in making demos of video prediction, please watch our latest update. Thanks again for your contribution!

**Comment by josephrocca** [comment]
Great to hear! I'm especially interested in any diffusion experiments you do. I did have a play around and it seems to help with 'stability' of the video, but it was getting "stuck" on frames, and I didn't have enough knowledge of SimVP or diffusion to really know what was going on.

* https://josephrocca.github.io/SimVP-web/demo/diffusion.html
* https://colab.research.google.com/gist/josephrocca/eccbb3e5ff92b8326d7e2627827b6323/simvp2-diffusion.ipynb
* https://gist.githubusercontent.com/josephrocca/2f67b18eb31c6c27924e6a159ff4c758/raw/ee7b36bd487e1d0a8315fa28344bcaeefb9997bc/CustomVideoDataset-diffusion.py

I am probably making some sort of silly mistake in there somewhere.

Anyway, looking forward to see your future results and improvements to SimVP!

**Comment by chengtan9907** [comment]
I will close this issue if there is no more question. Please feel free to open a new issue when you have new questions.
